@xen-orchestra/lite/src/components/vm/network/VmVifsTable.vue

```diff
@@ -0,0 +1,289 @@
+import {
+  faAlignLeft,
+  faAt,
+  faCaretDown,
+  faEdit,
+  faEllipsis,
+  faHashtag,
+  faPlus,
+  faPowerOff,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons'
+import { noop } from '@vueuse/shared'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const { vifs } = defineProps<{
+  vifs: XenApiVif[]
+}>()
+
+const { isReady, hasError } = useVifStore().subscribe()
+const { getByOpaqueRef: getNetworkByOpaqueRef } = useNetworkStore().subscribe()
+const { getByOpaqueRef: getGuestMetricsByOpaqueRef } = useVmGuestMetricsStore().subscribe()
+const { getByOpaqueRef: getVmByOpaqueRef } = useVmStore().subscribe()
+
+const { t } = useI18n()
+
+const selectedVifId = useRouteQuery('id')
+
+const vifsUuids = computed(() => vifs.map(vif => vif.uuid))
+
+const { selected, areAllSelected } = useMultiSelect(vifsUuids)
+
+const getNetworkName = (networkRef: XenApiNetwork['$ref']) => getNetworkByOpaqueRef(networkRef)?.name_label
+
+const getIpAddresses = (vif: XenApiVif) => {
+  const vm = getVmByOpaqueRef(vif.VM)
+
+  if (!vm) return []
+
+  const guestMetrics = getGuestMetricsByOpaqueRef(vm.guest_metrics)
+
+  if (!guestMetrics?.networks) return []
+
+  return [...new Set(Object.values(guestMetrics.networks).sort())]
+}
+
+const searchQuery = ref('')
+
+const filteredVifs = computed(() => {
+  const searchTerm = searchQuery.value.trim().toLocaleLowerCase()
+
+  if (!searchTerm) {
+    return vifs
+  }
+
+  return vifs.filter(vif =>
+    [...Object.values(vif), getNetworkName(vif.network)].some(value =>
+      String(value).toLocaleLowerCase().includes(searchTerm)
+    )
+  )
+})
+
+const { visibleColumns, rows } = useTable('vifs', filteredVifs, {
+  rowId: record => record.uuid,
+  columns: define => [
+    define('checkbox', noop, { label: '', isHideable: false }),
+    define('network', record => getNetworkName(record.network), { label: t('network') }),
+    define('device', record => t('vif-device', { device: record.device }), { label: t('device') }),
+    define('status', record => (record.currently_attached ? 'connected' : 'disconnected'), { label: t('status') }),
+    define('IP', record => getIpAddresses(record), { label: t('ip-addresses') }),
+    define('MAC', { label: t('mac-addresses') }),
+    define('MTU', { label: t('mtu') }),
+    define('locking_mode', { label: t('locking-mode') }),
+    define('more', noop, { label: '', isHideable: false }),
+  ],
+})
+
+type VifHeader = 'network' | 'device' | 'status' | 'IP' | 'MAC' | 'MTU' | 'locking_mode'
+
+const headerIcon: Record<VifHeader, IconDefinition> = {
+  network: faAlignLeft,
+  device: faAlignLeft,
+  status: faPowerOff,
+  IP: faAt,
+  MAC: faAt,
+  MTU: faHashtag,
+  locking_mode: faCaretDown,
+}
+</script>
+
+<style scoped lang="postcss">
+.vm-vif-table,
+.table-actions,
+.container {
+  display: flex;
+  flex-direction: column;
+}
+
+.vm-vif-table {
+  gap: 2.4rem;
+
+  .container,
+  .table-actions {
+    gap: 0.8rem;
+  }
```

Comment:
Based on our guidelines, we should nest the CSS:
```suggestion
.vm-vif-table {
  display: flex;
  flex-direction: column;
  gap: 2.4rem;

  .container,
  .table-actions {
    display: flex;
    flex-direction: column;
    gap: 0.8rem;
  }
```
However, in that case, I wonder what version is the best/most readable

Reply:
After trying I prefer yours